src/app/dataset-view/additional-components/lineage-component/services/lineage-graph-builder.service.ts

```diff
@@ -0,0 +1,145 @@
+import { Injectable } from "@angular/core";
+import { Edge, Node } from "@swimlane/ngx-graph";
+import { Observable, Subject, map } from "rxjs";
+import { DatasetLineageBasicsFragment, DatasetKind, FetchStepUrl } from "src/app/api/kamu.graphql.interface";
+import { LineageUpdate } from "src/app/dataset-view/dataset.subscriptions.interface";
+import { DatasetSubscriptionsService } from "src/app/dataset-view/dataset.subscriptions.service";
+import { LineageGraphNodeKind, LineageGraphNodeData, LineageNodeAccess } from "../lineage-model";
+
+@Injectable({
+    providedIn: "root",
+})
+export class LineageGraphBuilderService {
```

Comment:
There should be an entry point like `buildGraph`. Building nodes and edges should be private

Reply:
Fixed, added one public method `buildGraph`